### ISSUE TITLE
Wire generates code for protos that it shouldn't when using `root(...)` syntax

### ISSUE THREAD

**Issue by staktrace** [square/wire#2150]
Minimal reproducible testcase can be found here (sorry, it's square-private, since it may contain proprietary information): https://github.com/squareup/kartikaya/tree/main/wire-bug-20211125-jarfiles

In this testcase, there is a source jar file with two `.proto` files and a `root(...)` directive in the wire configuration that refers to a class in one of the `.proto` files. The expectation is that only that proto file is processed, since it doesn't refer to the other one at all. Instead, Wire generates code for both proto files and they get compiled and included in the produced jar file. This is undesirable becauase it is unnecessary work; the `root(...)` directive is supposed to limit the stuff that Wire processes but it doesn't seem to be working in this instance.

**Comment by staktrace** [comment]
I made a further reduced testcase for this, in the form of a test against the wire gradle-plugin. It can be found at https://github.com/staktrace/wire/commit/4da5fb606baead0b33ed023181a1cd1bc7d56a0c. In doing this reduction I discovered that a key aspect of the bug is that the .proto file that gets generated is extending some google protobuf thing. I'm not familiar with the sematics of this type of extension so I don't know if this behaviour is expected or not, but at least it's not as widespread as I originally thought.
